sql/expression/function/aggregation/unary_agg_buffers.go

```diff
@@ -417,12 +417,29 @@ func (c *countDistinctBuffer) Update(ctx *sql.Context, row sql.Row) error {
 		value = val
 	}
 
-	hash, err := hashstructure.Hash(value, nil)
-	if err != nil {
-		return fmt.Errorf("count distinct unable to hash value: %s", err)
+	var str string
+	for _, val := range value.(sql.Row) {
+		if val == nil {
+			return nil
```

Comment:
test for this case?

Reply:
and/or comment confirming that any nil in a row won't contribute to distinct count